### PR TITLE
Switch ON blobpool & ensure Cancun hardfork can occur

### PR DESCRIPTION
**Pull request by emailtovamos**
### Description

This PR switches ON blobpool and Cancun hardfork happens without error.

I had to set the WithdrawalsHash as non-nil, otherwise there was some mismatch during rlp encoding-decoding at least for me.

### CODE REVIEW

File: eth/handler.go [hunk 1]
```diff
@@ -702,8 +702,8 @@ func (h *handler) Start(maxPeers int, maxPeersPerIP int) {
 	// announce local pending transactions again
 	h.wg.Add(1)
 	h.reannoTxsCh = make(chan core.ReannoTxsEvent, txChanSize)
-	h.reannoTxsSub = h.txpool.SubscribeReannoTxsEvent(h.reannoTxsCh)
```
[the remark this answers is not in the document]
Comment (emailtovamos) [reply eth/handler.go:705]: This is not supported currently as can be seen here: https://github.com/bnb-chain/bsc/blob/blobtx/core/txpool/blobpool/blobpool.go#L1505

I am not sure if we should do it the same way that is done for normal transactions.

[the remark this answers is not in the document]
Comment (GalaIO) [reply eth/handler.go:705]: How about adding a TODO to describe the reason, and just returning `nil` in blob pool `SubscribeReannoTxsEvent` function?

File: consensus/parlia/parlia.go [hunk 2]
```diff
@@ -1223,6 +1223,15 @@ func (p *Parlia) FinalizeAndAssemble(chain consensus.ChainHeaderReader, header *
 		}
 	}
 
+	// consider withdrawals
+	shanghai := chain.Config().IsShanghai(header.Number, header.Time)
+	if shanghai {
+		// All blocks after Shanghai must include a withdrawals root.
+		if header.WithdrawalsHash == nil {
+			header.WithdrawalsHash = new(common.Hash)
```
Comment (GalaIO) [inline consensus/parlia/parlia.go:1231]: The `WithdrawalsHash` decode issue will be fixed by another PR. And you cannot change Shanghai's header field value, because it has been enabled on mainnet.

**Review by 0xbundler — APPROVED** [review]

**Review by GalaIO — APPROVED** [review]
